docs/docs/reference/debugger/debugger_repl.md

```diff
@@ -0,0 +1,388 @@
+
+#### `delete [Opcode]` (or shorthand `d [Opcode]`)
+
+Deletes a breakpoint at an opcode location. Usage is analogous to [the `break` command](#).
+
+### Variable inspection
+
+#### vars
+
+Show variable values available at this point in execution.
+
+Note: that the ability to inspect variable values from the debugger depends on compilation to be run in a special debug instrumentation mode. This instrumentation weaves variable tracing code with the original source code. 
+
+So variable value inspection comes at the expense of making the resulting ACIR bytecode bigger and harder to understand and optimize.
+
+If you find this compromise unacceptable, you can run the debugger with the flag `--skip-debug-instrumentation`. This will compile your circuit without any additional debug information, so the resulting ACIR bytecode will be identical to the one produced by standard Noir compilation. However, if you opt for this, the `vars` command will not be available while debugging.
+
+### Stacktrace
+
+#### `stacktrace`
+
+Displays the current stack trace.
+
+
+### Witness map
+
+#### `witness` (w)
+
+Show witness map. For example:
+
+```
+_0 = 0
+_1 = 2
+_2 = 1
+```
+
+#### `witness [Witness Index]`
+
+Display a single witness from the witness map. For example:
+
+```
+> witness 1
+_1 = 2
+```
+
+#### `witness [Witness Index] [New value]`
+
+Overwrite the given index with a new value. For example:
+
+```
+> witness 1 3
+_1 = 3
+```
+
+
+### Brillig registers
+
+#### `registers`
+
+Show Brillig registers. For example:
+
+```
+> registers
+0 = 0
+1 = 0
+2 = 0
+```
+
+Note: this command is only functional while the debugger is executing a Brillig block. 
+
+#### `regset [Register index] [New value]`
+
+Update a Brillig register with the given value. For example: 
+
+```
+> registers
+0 = 0
+1 = 0
+2 = 0
+> regset 1 10
+> regis
+0 = 0
+1 = 10
+2 = 0
+> 
+```
+
+### Brillig memory
+
+#### `memory`
+
+Show Brillig memory state. For example:
+
+```
+> memory
+At opcode 1.13: Store { destination_pointer: RegisterIndex(0), source: RegisterIndex(3) }
+...
+> registers
+0 = 0
+1 = 10
+2 = 0
+3 = 1
+4 = 1
+5 = 2³²
+6 = 1
+> into
+At opcode 1.14: Const { destination: RegisterIndex(5), value: Value { inner: 1 } }
+...
+> memory
+0 = 1
+>
+```
+
+In the example above: we start with a clean Brillig memory, then step through a `Store` opcode which stores the value of register 3 (1) into the memory address stored in register 0 (0). Thus now `memory` shows memory address 0 contains value 1.
+
+Note: this command is only functional while the debugger is executing a Brillig block.
+
+#### `memset [Memory address] [New value]`
+
+Update a Brillig memory cell with the given value. For example:
+
+```
+> memory
+0 = 1
+> memset 0 2
+> memory
+0 = 2
+> memset 1 4
+> memory
+0 = 2
+1 = 4
+>
+```
+
+Note: this command is only functional while the debugger is executing a Brillig block.
```

Comment:
Same for admonition blocks for the rest of the page.

Another thing to note: we are planning to have reference documentation be auto-generated with CLAP. This is on my TODO list. Would be much appreciated, if you're comfortable with CLAP, to add these docs straight into the debugger code. Otherwise just leave it and I'll find my way somehow 😄

Reply:
Re: CLAP, will address on a separate PR